source/common/config/datasource.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include "envoy/api/v2/core/base.pb.h"
+
+namespace Envoy {
+namespace Config {
+
+// envoy::api::v2::core::DataSource helper.
+class DataSource {
+public:
+  DataSource(const envoy::api::v2::core::DataSource& source) : source_(source) {}
```

Comment:
Given how this class is used, the API is a bit weird: every use constructs a temporary, calls one method, then destructs it.  Instead, should we just have two static functions here, that both take a DataSource& ?

Reply:
I agree with Greg, this feels weird. Is there any reason those are not just namespaced functions that take `source` as a parameter, similarly to those in `Envoy::Filesystem`?

Reply:
Was assuming the compiler would be smart enough to optimize any overhead here away, since the class is entirely const/immutable everywhere, but sure, I can make it just some namespace functions if you folks prefer this convention.